joss/paper.md

```diff
@@ -0,0 +1,163 @@
+energy-resolved X-ray pulsations. X-PSI provides a framework for the
+implementation of custom models (likelihood and prior functions) and for
+feeding those models to open-source statistical sampling software for use on
+high-performance computing systems. X-PSI supplies modules for post-processing
+posterior sample sets, and supplies tools for visualisation. For example, one
+can generate time- and energy-resolved (animated) images of model X-ray pulsars
+by tracing radiation from the stellar surface to a distant observer (such as a
+space telescope) as it propagates through spacetime; a snapshot of such an
+animation may be found in \autoref{fig:animation_snapshot}. Posterior summaries
+can be plotted of the time- and energy-domain signals that a model pulsar is
+inferred to generate, conditioned on observational data.
+
+![A snapshot from a time- and energy-resolved animation of a toy neutron star
+that generates X-ray pulsations. The top three panels are sky maps of photon
+specific intensity at three representative photon energies, and the bottom
+panels show instantaneous integrals over the sky maps, together with sky map
+integrals at past times normalized to the maximum at each photon energy
+(bottom-left panel) and additional photon energies (bottom-right
+panel).\label{fig:animation_snapshot}](_skymap_with_pulse_profile_and_spectrum_plot.png){width=100%}
+
+Notably, X-PSI has and is being applied by the NICER collaboration for
+pulse-profile modeling of rotation-powered millisecond pulsars. The
+publications that have directly applied the X-PSI package thus far are
+@Riley:2019 and @Riley:2021. Many more publications have used the accompanying
+open-source analysis pipeline and products published on Zenodo; these products
+may be accessed using the linked DOIs of @Riley:2019:Zenodo and
+@Riley:2021:Zenodo. The first of such publications were @Raaijmakers:2019 and
+@Bilous:2019, respectively on the topics of dense matter inference and
+multipolar magnetic fields.
+
+The numerical likelihood routines native to X-PSI are written in Cython
+[@cython2011], and are dependent on the GNU Scientific Library (GSL;
+@Gough:2009). High-level object-oriented model construction is performed by a
+user in the Python language, as is the interfacing with sampling software.
+Low-level customisation is encouraged in the extensions (either directly in
+Cython or via calls to external C libraries).  X-PSI is Unix source code
+compatible, and release versions are freely available on GitHub under the MIT
+license.
+
+*Software:* Python/C language [@Python2007], GNU Scientific Library (GSL;
+@Gough:2009), NumPy [@Numpy2011], Cython [@cython2011], OpenMP [@openmp], MPI
+for Python [@mpi4py], Matplotlib [@Hunter:2007; @matplotlibv2], IPython
+[@IPython2007], Jupyter [@Kluyver:2016aa], MultiNest [@MultiNest_2009],
+PyMultiNest [@PyMultiNest], GetDist [@Lewis19], nestcheck
+[@higson2018nestcheck;@higson2018sampling;@higson2019diagnostic], fgivenx
+[@fgivenx].
```

Comment:
It never occurred to me, but does X-PSI not rely on astropy at all? How do you read fits files?

Reply:
We don't read FITS files, our data comes in a text file!  So no need to use astropy.